src/main/native/windows/file.cc

```diff
@@ -446,20 +446,97 @@ int DeletePath(const wstring& path, wstring* error) {
     // DeleteFileW failed with access denied, but the path exists.
     if (attr & FILE_ATTRIBUTE_DIRECTORY) {
       // It's a directory or a junction.
-      if (!RemoveDirectoryW(wpath)) {
+
+      // Sometimes a deleted directory lingers in its parent dir
+      // after the deleting handle has already been closed.
+      // In this case we check the content of the parent directory,
+      // if we don't find any valid file, we try to delete it again after 5 ms.
+      // But we don't want to hang infinitely because another application
+      // can hold the handle for a long time. So try at most 20 times,
+      // which means a process time of 100-120ms.
+      // Inspired by https://github.com/Alexpux/Cygwin/commit/28fa2a72f810670a0562ea061461552840f5eb70
+      int count = 20;
+      while (count > 0) {
+        if (RemoveDirectoryW(wpath)) {
+          break;
+        }
         // Failed to delete the directory.
         err = GetLastError();
         if (err == ERROR_SHARING_VIOLATION || err == ERROR_ACCESS_DENIED) {
           // The junction or directory is in use by another process, or we have
           // no permission to delete it.
           return DeletePathResult::kAccessDenied;
-        } else if (err == ERROR_DIR_NOT_EMPTY) {
-          // The directory is not empty.
-          return DeletePathResult::kDirectoryNotEmpty;
         } else if (err == ERROR_FILE_NOT_FOUND || err == ERROR_PATH_NOT_FOUND) {
           // The directory or one of its directories disappeared or is no longer
           // a directory.
           return DeletePathResult::kDoesNotExist;
+        } else if (err == ERROR_DIR_NOT_EMPTY) {
+          static const wstring kDot(L".");
+          static const wstring kDotDot(L"..");
+          bool found_valid_file = false;
+          bool found_pending_delete_file = false;
+          WIN32_FIND_DATAW metadata;
+          HANDLE handle = ::FindFirstFileW((winpath + L"\\*").c_str(), &metadata);
+          if (handle != INVALID_HANDLE_VALUE) {
+            do {
+              if (kDot != metadata.cFileName && kDotDot != metadata.cFileName) {
+                std::wstring wname = winpath + L"\\" + metadata.cFileName;
```

Comment:
`wname` looks confusingly similar to `wpath` and `winpath`.
How about calling it `child` instead?

Reply:
Done.